cpp/oneapi/dal/algo/svm/detail/train_ops.cpp

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/backend/dispatcher.hpp"
+#include "oneapi/dal/algo/svm/detail/train_ops.hpp"
+#include "oneapi/dal/algo/svm/backend/cpu/train_kernel.hpp"
+
+namespace oneapi::dal::svm::detail {
+
+template <typename Float, typename Task, typename Method>
+struct train_ops_dispatcher<default_execution_context, Float, Task, Method> {
+    train_result operator()(const default_execution_context& ctx,
+                            const descriptor_base& desc,
+                            const train_input& input) const {
+        using kernel_dispatcher_t = dal::backend::kernel_dispatcher<
+            backend::train_kernel_cpu<Float, Task, Method>>;
+        return kernel_dispatcher_t()(ctx, desc, input);
+    }
+};
+
+#define INSTANTIATE(F, T, M) \
+  template struct train_ops_dispatcher<default_execution_context, F, T, M>;
```

Comment:
I think it's time to make a common header with this define.

Reply:
But it in namespace svm and I think for other algorithm it can change. For example PCA:
```c++
#define INSTANTIATE(F, M) \
  template struct train_ops_dispatcher<default_execution_context, F, M>;
```

Reply:
The only difference is presence of a task parameter. I think your define will be used in the same way for all algorithms we are implementing now. It's better to avoid copy-paste.

Reply:
ok, let's do the next pull request.